src/types/mutex.rs

```diff
@@ -0,0 +1,86 @@
+use std::ops::Deref;
+
+/// Wrapper for [`PyMutex`](https://docs.python.org/3/c-api/init.html#c.PyMutex) exposing an RAII interface.
+#[derive(Debug)]
+pub struct PyMutex<T> {
+    _mutex: crate::ffi::PyMutex,
```

Comment:
This design would permit moving the mutex while it's unlocked. As far as the C API docs state, they just say the mutex must never be moved. @colesbury is it permissible to move the mutex while it's unlocked?

Reply:
Yeah, it's the same restrictions as `pthread_mutex_t`. I think that's fine if Rust allows it for `std::sync::mutex`.

Reply:
I read the C api documentation as "once you create/use this, you can never move it again", which is why I pushed for the `PhantomPinned`. If it really means "Instances of PyMutex should not be copied or moved *while it is locked*.", we can remove that.

Also, while I appreciate your clarification here, would you mind adding it to https://docs.python.org/3.14/c-api/init.html#c.PyMutex ? I would feel a lot better if I can point to official documentation rather than a comment on a PR.